src/_guides/openapi/augmenting-generated-openapi.md

```diff
@@ -0,0 +1,349 @@
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Upload'
+```
+
+With that flag popped in we now have something to filter on using the sensibly named [openapi-filter](https://github.com/Mermade/openapi-filter/).
+
+```
+$ npm install --global openapi-filter
+
+$ openapi-filter --flags x-internal -- api/openapi.yaml api/openapi.public.yaml
+```
+
+Running the NPM script will filter out anything that has the flags specified, so this should removed the whole Upload operation from the source document called `openapi.yaml`, and shove it in a new document called `openapi.public.yaml`. Now I can deploy from this filtered version and the Uploads endpoint will be gone.
+
+```
+$ bump deploy api/openapi.public.yaml -d <DOC-ID> -t <TOKEN>
+* Your new documentation version will soon be ready at https://bump.sh/green-turtle/doc/tree-tracker-api
+
+Let's deploy a new version to your <DOC-ID> documentation on Bump.sh... done
+```
+
+Take a look in there and there's no more Upload! Phew. Right, with that sorted how can I get this working on every push? I've already set up a `.github/workflows/bump.sh` which runs on any pushes to main, so I can modify that a bit to use this new command:
+
+```yaml
+name: Deploy API documentation
+
+# snipped standard Bump deploy GitHib Action...
+
+jobs:
+  deploy-doc:
+    if: ${{ github.event_name == 'push' }}
+    name: Deploy API documentation on Bump.sh
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # added this
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: lts
+          cache: npm
+
+      # and this 
+      - name: Filter internal endpoints
+        run: |
+          npx openapi-filter -- --flags x-internal api/openapi.yaml api/openapi.public.yaml
+
+      - name: Deploy API documentation
+        uses: bump-sh/github-action@v1
+        with:
+          doc: <DOC-ID>
+          token: ${{secrets.BUMP_TOKEN}}
+          file: api/openapi.public.yaml # 👈 updated this
+```
+
+Now whenever anyone commits it'll publish the automatically created filtered `openapi.public.yaml` instead of the original version, and we won't even have to think about it again.
+
+You could also use this approach to deploy _two different_ sets of documentation, maybe on different [Bump.sh Hubs](https://bump.sh/api-catalog) for different teams with different permissions. 🙌
+
+**Another scenario:** what if we have been adding some new endpoints which aren't quite ready yet? Bump.sh users can use the `x-beta: true` [vender extension](/help/specification-support/doc-beta/) and filter that just as you would with `x-internal`. Another approach would be to use tags to mark things as beta, because openapi-filter will let you use tags as flags with the `--checkTags` flag. Here we can look for the `Beta` tag, and filter out any operations that use it.
```

Comment:
```suggestion
**Another scenario:** what if we have been adding some new endpoints which aren't quite ready yet? Bump.sh users can use the `x-beta: true` [vendor extension](/help/specification-support/doc-beta/) and filter that just as you would with `x-internal`. Another approach would be to use tags to mark things as beta, because openapi-filter will let you use tags as flags with the `--checkTags` flag. Here we can look for the `Beta` tag, and filter out any operations that use it.
```